src/gt4py/next/program_processors/runners/dace_iterator/workflow.py

```diff
@@ -0,0 +1,141 @@
+from gt4py.next.type_system import type_translation as tt
+
+
+@dataclasses.dataclass(frozen=True)
+class DaCeTranslator(
+    workflow.ChainableWorkflowMixin[
+        stages.ProgramCall,
+        stages.ProgramSource[languages.JSON, languages.LanguageSettings],
+    ],
+    step_types.TranslationStep[languages.JSON, languages.LanguageSettings],
+):
+    auto_optimize: bool = False
+    lift_mode: LiftMode = LiftMode.FORCE_INLINE
+    device_type: core_defs.DeviceType = core_defs.DeviceType.CPU
+    temporary_extraction_heuristics: Optional[
+        Callable[[itir.StencilClosure], Callable[[itir.Expr], bool]]
+    ] = None
+    use_field_canonical_representation: bool = False
+
+    def _language_settings(self) -> languages.LanguageSettings:
+        return languages.LanguageSettings(
+            formatter_key="",
+            formatter_style="",
+            file_extension="sdfg",
+        )
+
+    def __call__(
+        self,
+        inp: stages.ProgramCall,
+    ) -> stages.ProgramSource[languages.JSON, LanguageSettings]:
+        """Generate DaCe SDFG (JSON file) from the ITIR definition."""
+        program: itir.FencilDefinition = inp.program
+        on_gpu = True if self.device_type == core_defs.DeviceType.CUDA else False
+
+        # ITIR parameters
+        column_axis: Optional[Dimension] = inp.kwargs.get("column_axis", None)
+        offset_provider = inp.kwargs["offset_provider"]
+
+        sdfg = build_sdfg_from_itir(
+            program,
+            *inp.args,
+            offset_provider=offset_provider,
+            auto_optimize=self.auto_optimize,
+            on_gpu=on_gpu,
+            column_axis=column_axis,
+            lift_mode=self.lift_mode,
+            load_sdfg_from_file=False,
+            save_sdfg=False,
+            use_field_canonical_representation=self.use_field_canonical_representation,
+        )
+
+        arg_types = tuple(
+            interface.Parameter(param, tt.from_value(arg))
+            for param, arg in zip(sdfg.arg_names, inp.args)
+        )
+
+        module: stages.ProgramSource[languages.JSON, languages.LanguageSettings] = (
+            stages.ProgramSource(
+                entry_point=interface.Function(program.id, arg_types),
+                source_code=sdfg.to_json(),
+                library_deps=tuple(),
+                language=languages.JSON,
+                language_settings=self._language_settings(),
+            )
+        )
+        return module
+
+
+@dataclasses.dataclass(frozen=True)
+class DaCeCompiler(
+    workflow.ChainableWorkflowMixin[
+        stages.CompilableSource[SourceLanguageType, LanguageSettingsType, languages.Python],
+        stages.CompiledProgram,
+    ],
+    workflow.ReplaceEnabledWorkflowMixin[
+        stages.CompilableSource[SourceLanguageType, LanguageSettingsType, languages.Python],
+        stages.CompiledProgram,
+    ],
+    step_types.CompilationStep[SourceLanguageType, LanguageSettingsType, languages.Python],
```

Comment:
It would make sense to be more specific here and use `languages.SDFG_JSON` instead of `SourceLanguageType` and `languages.LanguageSettings` instead of `LanguageSettingsType`. Unless the `DaCeCompiler` can work with other languages (the one for gtfn can work with any generated language supported by CMake).

Reply:
I didn't check if there were other related occurrences

Reply:
Thanks, I'll fix this.